source/grid/tria.cc

```diff
@@ -15563,6 +15617,23 @@ const typename std::map<
 }
 
 
+template <int dim, int spacedim>
+DEAL_II_CXX20_REQUIRES((concepts::is_valid_dim_spacedim<dim, spacedim>))
+void Triangulation<dim, spacedim>::update_cell_relations()
+{
+  if (dynamic_cast<parallel::DistributedTriangulationBase<dim, spacedim> *>(
+        this))
+    return;
```

Comment:
You made this function no longer `virtual`, so the only way the `if` is `true` is if you have a parallel class but you explicitly call the function in the base class. Can this happen? If so, can you add a comment in which context this happens and why?